.ci/new_ci.jenkinsfile

```diff
@@ -0,0 +1,435 @@
+        return findPR[0][2]
+    }
+}
+
+def readEnvValues(String envFile) {
+    Map<String, String> envContent = [:]
+    String content = readFile(envFile)  // legacy/.env or modern/.env
+    echo content
+    content.split('\n').each({ it.trim() }).findAll({ it.size() && !it.startsWith('#') }).each {
+        def (name, value) = it.split('=')
+        echo "${name},${value}"
+        envContent[name] = value
+    }
+    return envContent
+}
+
+def parseVersion(String version) {
+    def matcher = (version =~ /(\d+).(\d+).(\d+)(\w{1}\d+)?/)
+    return [matcher[0][1] as String, matcher[0][2] as String, matcher[0][3] as String]
+}
+
+String getVersionMajor(String version) {
+    def (major, _1, _2) = parseVersion(version)
+    return major
+}
+
+def buildLegacyImage(Map image, String buildArgs, String registry, String registryCredentialsID) {
+    withCredentials([usernamePassword(credentialsId: registryCredentialsID, usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
+        sh 'docker login --username $USERNAME --password $PASSWORD ' + "${registry}"
+        sh 'docker pull docker.io/library/ubuntu:xenial'
+        sh 'docker pull docker.io/library/ubuntu:bionic'
+        sh 'docker pull docker.io/library/ubuntu:eoan'
+    }
+
+    // TODO: Add --cache-from value to speed-up the build
+
+    sh "docker build -t ${image["name"]} ${buildArgs} ."
+}
+
+def uploadImage(Map image, String registry, String crendentialsID) {
+    withCredentials([usernamePassword(credentialsId: crendentialsID, usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD')]) {
+        sh 'docker login --username $USERNAME --password $PASSWORD ' + "${registry}"
+    }
+
+    // Upload with the corresponding tag version
+    String uploadImage = "${registry}/${image["name"]}"
+    sh "docker tag ${image["name"]} ${uploadImage}"
+    sh "docker push ${uploadImage}"
+}
+
+def getLegacyImages(String pr_number, String registry) {
+    Map<String, String> envValues = [:]
+    stage('Legacy - Parse environment file') {
+        envValues = readEnvValues("legacy/.env")
+    }
+
+    List<Map> images = []
+    stage('Legacy - Generate build matrix') {
+        List<String> gccVersions = envValues["GCC_LEGACY_VERSIONS"].split(",")
+        List<String> clangVersions = envValues["CLANG_LEGACY_VERSIONS"] != null ? envValues["CLANG_LEGACY_VERSIONS"].split(",") : []
+        List<String> conanVersions = [envValues["CONAN_VERSION"]]
+        List<String> jenkinsAgentVersions = envValues["JENKINS_AGENT_VERSION"].split(",")
+        String dockerUsername = envValues["DOCKER_USERNAME"]
+
+        // <registry>/    <docker_username>/<compiler><majorVersion>(-jenkins):<conan_version>(-<jenkins_agent_version>)
+        List<String> compilerImages = []
+        gccVersions.each({ String gccVersion ->
+            String image = "${dockerUsername}/gcc${gccVersion}"
+            compilerImages.add(image)
+        })
+
+        clangVersions.each({ String clangVersion ->
```

Comment:
Don't need to check clang, we no longer build it and have no intention to do it.